clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -517,7 +517,12 @@ void SYCL::fpga::BackendCompiler::ConstructJob(
   if (Arg *FinalOutput = Args.getLastArg(options::OPT_o, options::OPT__SLASH_o,
                                          options::OPT__SLASH_Fe)) {
     SmallString<128> FN(FinalOutput->getValue());
-    FN.append(".prj");
+    StringRef ext = llvm::sys::path::extension(FN);
+    if (ext == ".o" || ext == ".a" || ext == ".out" || ext == ".obj" ||
+        ext == ".lib" || ext == ".exe")
```

Comment:
Overall looks good.  Nit - maybe use an array and find here instead?

Reply:
Done.